init/bootstrap.go

```diff
@@ -43,10 +46,18 @@ outer:
 			continue
 		}
 
-		for _, b := range buffer {
-			if b != 0 {
-				log.Infof("%s not empty", dev)
-				continue outer
+		boot2docker := false
+
+		if strings.HasPrefix(string(buffer), boot2dockerMagic) {
```

Comment:
@ibuildthecloud []byte and string don't share the same memory, So this copies 1048576  bytes of data. The other option was the slice implementation that I had prevously